src/Calculator.jsx

```diff
@@ -0,0 +1,134 @@
+    Container,
+    Grid
+} from '@material-ui/core';
+import ReactDOM from "react-dom";
+import React from "react";
+
+import Header from './Header';
+import Variable from './Variable';
+import ResultBox from './ResultBox';
+import ErrorMessage from './ErrorMessage';
+
+import model from './model';
+
+const labels = require("./labels.json");
+
+document.title = labels.title + ": " + labels.info;
+
+export default function Calculator (props) {
+    const initialValues = {};
+    model.variables.map((v) => {initialValues[v.name] = 0 });
+
+    const [values, setValues] = React.useState(initialValues);
+    const [score, setScore] = React.useState(Math.NaN);
+    const [showScore, setShowScore] = React.useState(false);
+    const [error, setError] = React.useState({});
+    const [showError, setShowError] = React.useState(false);
+
+    function populateWithSampleData() {
+        setValues(model.sample_values);
+    }
+
+    function handleCloseScore() {
+        setShowScore(false);
+    }
+
+    function handleCloseError() {
+        setShowError(false);
+        for (var v in error) {
+            if (error[v]) {
+                console.log(v);
+                return;
+            }
+        }
+    }
+
+    function _isValueValid(value) {
+        return typeof(value) !== "undefinded" && value !== '' && !isNaN(value);
+    }
+
+    function _prepareValues() {
+        var data = [];
+        var hasError = false, errorState = {};
+        for (var i = 0; i < model.variables.length; i++) {
+            const fieldMeta = model.variables[i];
+            if (_isValueValid(values[fieldMeta.name])) {
+                data.push(values[fieldMeta.name]);
+            } else {
+                errorState[fieldMeta.name] = hasError = true;
+            }
+        }
+        setError(errorState);
+        if (hasError) {
+            setShowError(true);
+            return;
+        }
+        return data;
+    }
+
+    function compute() {
+        var data = _prepareValues()
+        if (data) {
+            var result = model.estimate(data);
+            setScore(result);
+            setShowScore(true);
+        } else {
+            setScore(Math.NaN);
+        }
+    }
+
+    function reset() {
+        setValues(initialValues);
+        setError({});
+    }
+
+    return (
+        <Container maxWidth="sm">
+             <Header text={labels.title} info={labels.info}/>
+             <Grid container direction="column" alignItems="stretch" justify="space-between" spacing={5}>
+                 {model.variables.map((variable, idx) => <Grid item key={"variable_container" + variable.name}>
+                                                               <Variable
+                                                                   value={values[variable.name]}
+                                                                   metadata={variable}
+                                                                   key = {"variable_" + variable.name}
```

Comment:
Spaces or no spaces around `=`? Pick one. This is inconsistent in most files.

Reply:
I find that if all attributes are on the same row `attr="value"` works better, while on multiple rows `attr = "value"` is easier on the eye. @sdumitriu objections?

Reply:
OK.